### PR TITLE
Feature/debt and expense api

### PR DESCRIPTION
**Pull request by shubhamsahu15011998**

### CODE REVIEW

File: src/main/java/com/technogise/expensesharingapp/repositories/ExpenseRepository.java [hunk 1]
```diff
@@ -2,10 +2,19 @@
...
+import java.util.List;
+
 @Repository
 @Component("expenseRepository")
 public interface ExpenseRepository extends JpaRepository<Expense, Long> {
+
+  @Query("SELECT e FROM Expense e WHERE e.payerId=?1 or " +
+      "e.id in (SELECT d.expenseId FROM ExpenseDebtor d WHERE d.debtorId=?1) " +
+      "ORDER BY e.createdAt")
+  List<Expense> getAllExpensesById(Long userId);
```
Comment (chandan911) [inline src/main/java/com/technogise/expensesharingapp/repositories/ExpenseRepository.java]: `findAllExpensesByUserId`

File: src/main/java/com/technogise/expensesharingapp/repositories/DebtRepository.java [hunk 2]
```diff
@@ -2,10 +2,17 @@
...
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 @Component("debtRepository")
 public interface DebtRepository extends JpaRepository<Debt, Long> {
+
+  @Query("SELECT d FROM Debt d WHERE d.creditorId=?1 or d.debtorId=?1 ORDER BY d.createdAt")
+  List<Debt> getAllDebtsByUserId(Long userId);
```
Comment (chandan911) [inline src/main/java/com/technogise/expensesharingapp/repositories/DebtRepository.java]: `findAllDebtsByUserId `

File: src/main/java/com/technogise/expensesharingapp/responseModels/AggregateDataResponse.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  private List<DebtResponse> debts;
+
+  private User user;
+
+  private List<User> otherUser;
+
+  public AggregateDataResponse() {
+  }
+
+  public AggregateDataResponse(List<ExpenseResponse> expenses, List<DebtResponse> debts, User user, List<User> otherUser) {
```
Comment (chandan911) [inline src/main/java/com/technogise/expensesharingapp/responseModels/AggregateDataResponse.java]: `otherUsers`

File: src/main/java/com/technogise/expensesharingapp/util/ResponseGeneratorImpl.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    debtResponse.setDebtor(userService.getUserById(debt.getDebtorId()).get().getName());
+    return debtResponse;
+  }
+
+  @Override
+  public AggregateDataResponse aggregateResponseGenerator
+      (List<Expense> expenses, List<Debt> debts, User user, List<User> allUsers) {
+    List<ExpenseResponse> expenseResponses = new ArrayList<ExpenseResponse>();
+    for (int index = 0; index < expenses.size(); index++)
+      expenseResponses.add(expenseResponseGenerator(expenses.get(index)));
```
Comment (chandan911) [inline src/main/java/com/technogise/expensesharingapp/util/ResponseGeneratorImpl.java]: avoid side effect

expenses.map(expense -> expenseResponseGenerator(expense))
expenses.foreach(expense -> expenseResponses.add(expense)) // bad

File: src/test/java/com/technogise/expensesharingapp/controllers/UserControllerTest.java [hunk 5]
```diff
@@ -1,47 +1,66 @@
...
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 @SpringBootTest
 @ActiveProfiles("test")
 @AutoConfigureMockMvc
 public class UserControllerTest {
 
-  @MockBean
+  @Autowired
```
Comment (chandan911) [inline src/test/java/com/technogise/expensesharingapp/controllers/UserControllerTest.java]: To be fixed

**Review by chandan911 — APPROVED** [review]

**Review by satyam-saxen — APPROVED** [review]
I'm approving this PR. Just a tip - Generate setter only when required.

**Pull request merged**